base/printf.jl

```diff
@@ -1,1254 +1,830 @@
-end
-
-function decode_hex(d::Integer, symbols::AbstractArray{UInt8,1}, digits)
-    neg, x = handlenegative(d)
-    @handle_zero x digits
-    pt = i = (sizeof(x)<<1)-(leading_zeros(x)>>2)
-    while i > 0
-        digits[i] = symbols[(x&0xf)+1]
-        x >>= 4
-        i -= 1
+    if N > 8
+        for i = 9:length(f.formats)
+            pos, argp = fmt(buf, pos, args, argp, f.formats[i])
+            for j in f.substringranges[i + 1]
+                buf[pos] = f.str[j]
+                pos += 1
+            end
+        end
     end
-    return Int32(pt), Int32(pt), neg
+    return pos
 end
 
-const hex_symbols = b"0123456789abcdef"
-const HEX_symbols = b"0123456789ABCDEF"
-
-decode_hex(x::Integer, digits) = decode_hex(x,hex_symbols,digits)
-decode_HEX(x::Integer, digits) = decode_hex(x,HEX_symbols,digits)
 
-function decode(b::Int, x::BigInt, digits)
-    neg = x.size < 0
-    pt = Base.ndigits(x, base=abs(b))
-    length(digits) < pt+1 && resize!(digits, pt+1)
-    neg && (x.size = -x.size)
-    GMP.MPZ.get_str!(digits, b, x)
-    neg && (x.size = -x.size)
-    return Int32(pt), Int32(pt), neg
+function plength(f::Spec{T}, args, argp) where {T <: Chars}
+    width, prec, argp = dynamic(args, argp, f)
+    x = args[argp]
+    argp =+ 1
+    max(width, 1) + (ncodeunits(x isa AbstractString ? x[1] : Char(x)) - 1), argp
 end
-decode_oct(x::BigInt, digits) = decode(8, x, digits)
-decode_dec(x::BigInt, digits) = decode(10, x, digits)
-decode_hex(x::BigInt, digits) = decode(16, x, digits)
-decode_HEX(x::BigInt, digits) = decode(-16, x, digits)
 
-function decode_0ct(x::BigInt, digits)
-    neg = x.size < 0
-    digits[1] = '0'
-    if x.size == 0
-        return Int32(1), Int32(1), neg
-    end
-    pt = Base.ndigits0z(x, 8) + 1
-    length(digits) < pt+1 && resize!(digits, pt+1)
-    neg && (x.size = -x.size)
-    p = convert(Ptr{UInt8}, digits) + 1
-    GMP.MPZ.get_str!(p, 8, x)
-    neg && (x.size = -x.size)
-    return neg, Int32(pt), Int32(pt)
-end
-
-### decoding functions directly used by printf generated code ###
-
-# decode_*(x)=> fixed precision, to 0th place, filled out
-# fix_*(x,n) => fixed precision, to nth place, not filled out
-# ini_*(x,n) => n initial digits, filled out
-
-# alternate versions:
-#   *_0ct(x,n) => ensure that the first octal digits is zero
-#   *_HEX(x,n) => use uppercase digits for hexadecimal
-
-# - returns (len, point, neg)
-# - implies len = point
-#
-
-function decode_dec(x::SmallFloatingPoint, digits)
-    if x == 0.0
-        digits[1] = '0'
-        return (Int32(1), Int32(1), false)
-    end
-    len,pt,neg = grisu(x,Grisu.FIXED,0,digits)
-    if len == 0
-        digits[1] = '0'
-        return (Int32(1), Int32(1), false)
-    else
-        for i = len+1:pt
-            digits[i] = '0'
-        end
-    end
-    return Int32(len), Int32(pt), neg
+function plength(f::Spec{Pointer}, args, argp)
+    width, prec, argp = dynamic(args, argp, f)
+    x = args[argp]
+    argp =+ 1    
```

Comment:
There's some trailing space here (and elsewhere) to remove.

Reply:
(see something like `git diff origin/jq/printf...jmon/printf` to show the whitespace)